script/gitlabci/job_base.yml

```diff
@@ -145,3 +145,41 @@
   tags:
     - amd
     - rocm
+
+# -------- clang ---------
+
+.basecfg_clang:
+  image: ubuntu:${ALPAKA_CI_UBUNTU_VER}
+  variables:
+    ALPAKA_CI_UBUNTU_VER: "20.04"
+    CC: clang
+    CXX: clang++
+    ALPAKA_CI_SANITIZERS: ""
+    ALPAKA_CI_ANALYSIS: "OFF"
+    ALPAKA_CI_TBB_VERSION: 2021.4.0
+    ALPAKA_CI_RUN_TESTS: "ON"
+    alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE: "ON"
+    alpaka_ACC_CPU_B_SEQ_T_THREADS_ENABLE: "ON"
+    alpaka_ACC_CPU_B_TBB_T_SEQ_ENABLE: "ON"
+    alpaka_ACC_CPU_B_OMP2_T_SEQ_ENABLE: "ON"
+    alpaka_ACC_CPU_B_SEQ_T_OMP2_ENABLE: "ON"
+    alpaka_ACC_ANY_BT_OMP5_ENABLE: "OFF"
+    alpaka_ACC_ANY_BT_OACC_ENABLE: "OFF"
+    alpaka_ACC_GPU_CUDA_ENABLE: "OFF"
+    alpaka_ACC_GPU_CUDA_ONLY_MODE: "OFF"
+    alpaka_ACC_GPU_HIP_ENABLE: "OFF"
+    alpaka_ACC_GPU_HIP_ONLY_MODE: "OFF"
+  extends: .base
+
+.base_clang_legacy:
+  extends: .basecfg_clang
+  tags:
+    - cpuonly
+    - x86_64 # aarch64 only supported for clang versions >= 12
+
+.base_clang: 
+  extends: .basecfg_clang
+  variables:
+    ALPAKA_CI_CMAKE_VER: 3.19.8 # Don't go below 3.19.3, there are no CMake binaries for ARM
+  tags:
+    - cpuonly
```

Comment:
It's nearly random if a job basing on this base job is executed on x86 or ARM. This means also, that the CI can run all jobs on x86.

Reply:
Yes, randomness is what I'm going for here. We should reliably run across architectures.

Reply:
If we use the pair-wise generator, we don't need randomness anymore. The idea of the pair-wise generator is, that each pair of parameters occurs at least one time in the matrix. So if we put the processor architecture as additional parameter, the generator guaranties, that the each supported version of a compiler is tested with ARM.